vkcs/resource_vkcs_db_config_group.go

```diff
@@ -0,0 +1,215 @@
+package vkcs
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDatabaseConfigGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDatabaseConfigGroupCreate,
+		ReadContext:   resourceDatabaseConfigGroupRead,
+		UpdateContext: resourceDatabaseConfigGroupUpdate,
+		DeleteContext: resourceDatabaseConfigGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"datastore": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"version": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"values": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"datastore_version_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDatabaseConfigGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(configer)
+	DatabaseV1Client, err := config.DatabaseV1Client(getRegion(d, config))
+	if err != nil {
+		return diag.Errorf("Error creating VKCS database client: %s", err)
+	}
+
+	createOpts := dbConfigGroupCreateOpts{
+		Name: d.Get("name").(string),
+	}
+
+	v := d.Get("datastore")
+	datastore, err := extractDatabaseDatastore(v.([]interface{}))
+	if err != nil {
+		return diag.Errorf("unable to determine vkcs_db_config_group datastore")
+	}
+	createOpts.Datastore = &datastore
+
+	dsParameters, err := dbDatastoreParametersGet(DatabaseV1Client, datastore.Type, datastore.Version).extract()
```

Comment:
should not we put this parameter transformation to a separate func?

Reply:
where?

Reply:
Here: https://github.com/vk-cs/terraform-provider-vkcs/pull/25/files#diff-e53172a2e2319ef34885beb09cedc98b980a306a1efe94fdf99a36b4573d0b16R93